### PR TITLE
WIP Feature/account tab

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/components/chat/tabs/AccountTab.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          fromDate
+          title
+        }
+      }
+    }
+  }
+`
+
+export interface AccountTabProps {
+  showNotification: ({}) => void
```
Comment (palmfjord) [inline src/components/chat/tabs/AccountTab.tsx]: this looks weird

File: src/components/chat/tabs/AccountTab.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    query={GET_MEMBER_ACCOUNT_QUERY}
+    variables={{ memberId: props.match.params.id }}
+  >
+    {({ data, loading, error }) => {
+      if (!data || loading) {
+        return 'loading'
+      }
+      return (
+        <>
+          <h3>Current balance: {data.member.account.balance.amount} kr</h3>
```
Comment (palmfjord) [inline src/components/chat/tabs/AccountTab.tsx]: maybe we should pull in some proper money formatting?

File: src/components/chat/tabs/account-tab/add-entry-form.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                .then(() => {
+                  this.props.showNotification({
+                    message: 'Account entry added.',
+                    header: 'Success',
+                    type: 'olive',
+                  })
+                  resetForm()
+                })
+                .catch((error) => {
+                  this.props.showNotification({
```
Comment (palmfjord) [inline src/components/chat/tabs/account-tab/add-entry-form.tsx]: maybe we should throw again? just so we don't swallow the error

File: src/components/payouts/payout-details/index.tsx [hunk 4]
```diff
@@ -72,7 +72,12 @@ class PayoutDetails extends React.Component<
               <MuiMenuItem value="REFERRAL">Referral</MuiMenuItem>
               <MuiMenuItem value="REFUND">Refund</MuiMenuItem>
             </Field>
-            <Field component={TextField} label="Payout amount" name="amount" />
+            <Field
+              component={TextField}
+              label="Payout amount"
+              name="amount"
+              pattern="^\\d+$"
```
Comment (palmfjord) [inline src/components/payouts/payout-details/index.tsx]: 🤨

File: src/components/chat/tabs/account-tab/add-entry-form.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                    message: error.message,
+                    header: 'Error',
+                    type: 'olive',
```
Comment (palmfjord) [inline src/components/chat/tabs/account-tab/add-entry-form.tsx]: 🙈

File: src/components/chat/tabs/AccountTab.tsx (hunk 1 above)
Comment (vonElfvin) [reply src/components/chat/tabs/AccountTab.tsx]: Change to:
showNotification: ({message: string, header: string, type: string}) => void

maybe?

File: src/components/chat/tabs/account-tab/add-entry-form.tsx [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+                            {formatMoneySE({
+                              amount: parsedAmount,
+                              currency: 'SEK',
+                            })}{' '}
+                            <strong>less</strong>
+                          </>
+                        ) : (
+                          <>
+                            {this.props.firstName} will owe us{' '}
+                            {parsedAmount * -1} <strong>more</strong>
```
Comment (vonElfvin) [inline src/components/chat/tabs/account-tab/add-entry-form.tsx]: formatMoneySE here as well?

File: src/components/chat/tabs/AccountTab.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+              <TableBody>
+                {data.member.account.entries.map((entry) => (
+                  <TableRowColored key={entry.id} amount={entry.amount.amount}>
+                    <TableCell>{entry.fromDate}</TableCell>
+                    <TableCell>
+                      {entry.id}
+                      <br />
+                      {entry.title
+                        ? entry.title
+                        : `${entry.reference} (${entry.source})`}
```
Comment (vonElfvin) [inline src/components/chat/tabs/AccountTab.tsx]: Use a "entry.title ?: `${entry.reference} (${entry.source})`" here?

File: src/components/chat/tabs/account-tab/add-entry-form.tsx [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+      memberId
+    }
+  }
+`
+
+interface State {
+  confirmed: boolean
+}
+
+const SubmitButton = withStyles({
```
Comment (vonElfvin) [inline src/components/chat/tabs/account-tab/add-entry-form.tsx]: Change name to "Button" ?

**Pull request merged**